acquisitions/src/main/resources/thunderjet/mod-orders/features/check-encumbrance-status-after-moving-expended-value.feature

```diff
@@ -0,0 +1,189 @@
+    Then status 200
+    * def poLine = $
+    * def encumbranceId1 = poLine.fundDistribution[0].encumbrance
+    * def encumbranceId2 = poLine.fundDistribution[1].encumbrance
+
+    * print "Add an invoice line linked to the po line"
+    * copy invoiceLine = invoiceLineTemplate
+    * set invoiceLine.id = invoiceLineId
+    * set invoiceLine.invoiceId = invoiceId
+    * set invoiceLine.poLineId = poLineId
+    * set invoiceLine.fundDistributions[0] = { fundId:'#(fundId1)', code: 'fundCode1', encumbrance: '#(encumbranceId1)', distributionType:'percentage', value:50 }
+    * set invoiceLine.fundDistributions[1] = { fundId:'#(fundId2)', code: 'fundCode2', encumbrance: '#(encumbranceId2)', distributionType:'percentage', value:50 }
+    * set invoiceLine.total = 1
+    * set invoiceLine.subTotal = 1
+    * set invoiceLine.releaseEncumbrance = true
+    * remove invoiceLine.fundDistributions[0].expenseClassId
+    Given path 'invoice/invoice-lines'
+    And request invoiceLine
+    When method POST
+    Then status 201
+
+  Scenario: Approve the invoice
+    Given path 'invoice/invoices', invoiceId
+    When method GET
+    Then status 200
+    * def invoice = $
+    * set invoice.status = 'Approved'
+    Given path 'invoice/invoices', invoiceId
+    And request invoice
+    When method PUT
+    Then status 204
+
+  Scenario: Pay the invoice
+    Given path 'invoice/invoices', invoiceId
+    When method GET
+    Then status 200
+    * def invoice = $
+    * set invoice.status = 'Paid'
+    Given path 'invoice/invoices', invoiceId
+    And request invoice
+    When method PUT
+    Then status 204
+
+  Scenario: Update fundId in poLine
+    Given path 'orders/order-lines', poLineId
+    When method GET
+    Then status 200
+    * def poLine = $
+    * set poLine.fundDistribution[0].fundId = fundId3
+    * set poLine.fundDistribution[0].code = 'fundCode3'
+    * set poLine.fundDistribution[0].value = 30
+    * set poLine.fundDistribution[1].value = 30
+    * set poLine.fundDistribution[2] = { fundId:'#(fundId4)', code: 'fundCode4', distributionType:'percentage', value: 40 }
+    * remove poLine.fundDistribution[0].encumbrance
+
+    Given path 'orders/order-lines', poLineId
+    And request poLine
+    When method PUT
+    Then status 204
+
+  Scenario Outline: Check the newly created encumbrance
+    Given path 'orders/order-lines', poLineId
+    When method GET
+    Then status 200
+    * def newEncumbranceId = $.fundDistribution[<index>].encumbrance
+
+    Given path 'finance/transactions', newEncumbranceId
+    When method GET
+    Then status 200
+    And match $.amount == <amount>
+    And match $.encumbrance.amountExpended == <amountExpended>
+    And match $.encumbrance.status == <status>
+
+    Examples:
+      | index | amount | amountExpended | status       |
+      | 0     | 0      | 0.5            | 'Released'   |
+      | 1     | 0      | 0.5            | 'Released'   |
+      | 2     | 0.4    | 0              | 'Unreleased' |
```

Comment:
please add some comment or log etc to describe why these values are expected values

Reply:
commented

Reply:
thank you, not it is easier to understand